catch/include/resource_guards.hh

```diff
@@ -55,15 +75,36 @@ template <typename T> class LinearAllocGuard {
       case LinearAllocs::hipMallocManaged:
         HIP_CHECK(hipMallocManaged(reinterpret_cast<void**>(&ptr_), size, flags ? flags : 1u));
         host_ptr_ = ptr_;
+        break;
+      case LinearAllocs::noAlloc:
+        break;
     }
   }
 
   LinearAllocGuard(const LinearAllocGuard&) = delete;
-  LinearAllocGuard(LinearAllocGuard&&) = delete;
+
+  LinearAllocGuard(LinearAllocGuard&& o)
+      : allocation_type_{o.allocation_type_}, ptr_{o.ptr_}, host_ptr_{o.host_ptr_} {
+    o.allocation_type_ = LinearAllocs::noAlloc;
+    o.ptr_ = nullptr;
+    o.host_ptr_ = nullptr;
+  }
+
+  LinearAllocGuard& operator=(LinearAllocGuard&& o) {
+    allocation_type_ = o.allocation_type_;
+    ptr_ = o.ptr_;
```

Comment:
The `allocation_type` and `ptr_` need to be checked and freed if needed.

Reply:
Done.